packages/main/src/Dialog.js

```diff
@@ -295,6 +325,9 @@ class Dialog extends Popup {
 		this._isRTL = this.effectiveDir === "rtl";
 		this.onPhone = isPhone();
 		this.onDesktop = isDesktop();
+		if (this.state === ValueState.Error || this.state === ValueState.Warning) {
+			this.role = "alertdialog";
+		}
```

Comment:
This approach will not reset the role back to "dialog" when the state goes from "Error/Warning" to some other value like "Success".

Extract this to a getter that will always return "dialog" or "alertdialog" based on the current state and use it in the hbs file.